src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1299,6 +1328,28 @@ public void OpenFileFromPath(string filePath, bool forceManualExecutionMode = fa
             Logger.LogError("Could not open workspace at: " + filePath);
         }
 
+        static private DynamoPreferencesData DynamoPreferencesDataFromJson(string json)
+        {
+            JsonReader reader = new JsonTextReader(new StringReader(json));
+            var obj = JObject.Load(reader);
+            var viewBlock = obj["View"];
+            var dynamoBlock = viewBlock["Dynamo"];
```

Comment:
let's assume that for some graphs this value is null - we should still deserialize the rest of the graph okay, currently it looks like we'll throw immediately on calling ToString() below, and then in a bunch of places during OpenJSON file - can we check for null in all these places?

Reply:
Neither `View` nor `Dynamo` are required, and so we have to assume that they may be null.

Reply:
I have added code to handle both cases where either the view block is entirely missing, or there is a view block but no Dynamo preferences block in the view block.